mlir/lib/Dialect/GPU/Transforms/SerializeToHsaco.cpp

```diff
@@ -95,7 +95,7 @@ class SerializeToHsacoPass
   std::unique_ptr<std::vector<char>>
   serializeISA(const std::string &isa) override;
 
-  std::unique_ptr<SmallVectorImpl<char>> assembleIsa(const std::string &isa);
+  std::unique_ptr<SmallVector<char, 16>> assembleIsa(const std::string &isa);
```

Comment:
IMO this should either return `SmallVector<char>` (without an exact size) or take `SmallVectorImpl<char> &` as the second function argument

Reply:
I'm not familiar enough with this code to know what would be best, and honestly I didn't spend any time trying to figure that out. My goal was just to get it to compile with my SmallVectorImpl change. I hope that the maintainers of this code will make a change that will make my change unnecessary.

Reply:
@krzysz00 can you change this ahead of the current PR?